### ISSUE TITLE
Silence duplicate key warning barrage

### ISSUE THREAD

**Issue by atz** [sul-dlss/argo#572]
Looks like:
```
(eval):1: warning: key "exploded_tag_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "objectType_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "content_type_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "rights_primary_ssi" is duplicated and overwritten on line 1
(eval):1: warning: key "use_license_machine_ssi" is duplicated and overwritten on line 1
(eval):1: warning: key "nonhydrus_collection_title_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "hydrus_collection_title_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "nonhydrus_apo_title_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "hydrus_apo_title_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "current_version_isi" is duplicated and overwritten on line 1
(eval):1: warning: key "processing_status_text_ssi" is duplicated and overwritten on line 1
(eval):1: warning: key "released_to_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "wf_wps_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "wf_wsp_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "wf_swp_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "has_model_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "metadata_source_ssi" is duplicated and overwritten on line 1
(eval):1: warning: key "registered_dttsim" is duplicated and overwritten on line 1
(eval):1: warning: key "accessioned_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "published_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "submitted_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "deposited_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "modified_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "opened_latest_dttsi" is duplicated and overwritten on line 1
(eval):1: warning: key "embargo_release_dtsim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_format_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_pub_date_facet_ssi" is duplicated and overwritten on line 1
(eval):1: warning: key "topic_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_subject_geographic_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_subject_temporal_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_genre_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "sw_language_ssim" is duplicated and overwritten on line 1
(eval):1: warning: key "mods_typeOfResource_ssim" is duplicated and overwritten on line 1
```

Happens on basically each request.

**Comment by atz** [comment]
Warnings are triggered from inside this (pry by hand) trace:


https://github.com/rsolr/rsolr/blob/v1.0.13/lib/rsolr/client.rb#L287
```ruby
send "evaluate_#{request[:params][:wt]}_response", request, response
```

https://github.com/rsolr/rsolr/blob/v1.0.13/lib/rsolr/client.rb#L190
```ruby
adapt_response(request_context, raw_response) unless raw_response.nil?
```

https://github.com/rsolr/rsolr/blob/v1.0.13/lib/rsolr/client.rb#L190
```ruby
adapt_response(request_context, raw_response) unless raw_response.nil?
```

https://github.com/rsolr/rsolr/blob/v1.0.13/lib/rsolr/client.rb#L176
```ruby
execute request_context
```
https://github.com/projectblacklight/blacklight/blob/v5.9.4/lib/blacklight/solr_repository.rb#L48
```ruby
res = blacklight_solr.send_and_receive(path, {key=>solr_params.to_hash, method:blacklight_config.http_method})
```

**Comment by atz** [comment]
The solr response in question (unreadably huge, but at least very literal):
https://gist.github.com/atz/a85712d6529d4c5ee7fb

**Comment by atz** [comment]
So this is an artifact of using ruby as the response type and the "parsing" of that response being to directly eval the body.  It would go away completely if we used `:wt => 'json'` (where, notably, repeated keys in a hash are actually legal).

**Comment by atz** [comment]
If we care about the repetition possibly being a logical or configuration error, then we need to parse that massive gist and figure out what we actually mean by that.

**Comment by atz** [comment]
Duplication is present in Solr response, specifically the facet_fields:
```ruby
    'facet_fields' => {
      'exploded_tag_ssim' => ['Project',12,'Remediated By',10,'Registered By',8,'Project : DOR',6,'Remediated By : 3.25.3',6,'Project : Fuller Slides',3,'Registered By : renzo',3,'Remediated By : 4.6.6.2',3,'Process',2,'Process : Content Type',2,'Process : Content Type : Book (flipbook, ltr)',2,'Project : Argo Demo',2,'Registered By : llam813',2,'DPG',1,'DPG : Curator Request',1,'DPG : Curator Request : Kasianovitz',1,'DPG : State Banking Commission Annual Reports',1,'DPG : State Banking Commission Annual Reports : Virginia',1,'Process : Rights',1,'Process : Rights : world',1,'Project : State Banking Commission Annual Reports',1,'Registered By : labware',1,'Registered By : mbklein',1,'Registered By : sunetid',1,'Remediated By : 4.22.3',1],
      'objectType_ssim' => ['adminPolicy',7,'item',6,'agreement',2,'workflow',2,'collection',1],
      'content_type_ssim' => ['image',3,'file',2],
      'rights_primary_ssi' => ['dark',13,'stanford',3,'world',2],
      'use_license_machine_ssi' => [],
      'nonhydrus_collection_title_ssim' => ['Annual report of the State Corporation Commission showing the condition of the incorporated state banks and other institutions operating in Virginia at the close of business',1],
      'hydrus_collection_title_ssim' => [],
      'nonhydrus_apo_title_ssim' => ['Stanford University Libraries - Special Collections',4,'State Banking Commission Annual Reports',2,'[Internal System Objects]',12],
      'hydrus_apo_title_ssim' => [],
      'current_version_isi' => ['1',11,'2',4,'4',1],
      'processing_status_text_ssi' => ['Unknown Status',15,'In accessioning',1],
      'released_to_ssim' => [],
      'wf_wps_ssim' => ['accessionWF',1,'accessionWF:content-metadata',1,'accessionWF:content-metadata:waiting',1,'accessionWF:descriptive-metadata',1,'accessionWF:descriptive-metadata:ready',1,'accessionWF:descriptive-metadata:waiting',1,'accessionWF:end-accession',1,'accessionWF:end-accession:waiting',1,'accessionWF:provenance-metadata',1,'accessionWF:provenance-metadata:waiting',1,'accessionWF:publish',1,'accessionWF:publish:waiting',1,'accessionWF:remediate-object',1,'accessionWF:remediate-object:waiting',1,'accessionWF:reset-workspace',1,'accessionWF:reset-workspace:waiting',1,'accessionWF:rights-metadata',1,'accessionWF:rights-metadata:waiting',1,'accessionWF:sdr-ingest-received',1,'accessionWF:sdr-ingest-received:ready',1,'accessionWF:sdr-ingest-received:waiting',1,'accessionWF:sdr-ingest-transfer',1,'accessionWF:sdr-ingest-transfer:waiting',1,'accessionWF:shelve',1,'accessionWF:shelve:waiting',1,'accessionWF:start-accession',1,'accessionWF:start-accession:completed',1,'accessionWF:technical-metadata',1,'accessionWF:technical-metadata:waiting',1],
      'wf_wsp_ssim' => ['accessionWF',1,'accessionWF:completed',1,'accessionWF:completed:start-accession',1,'accessionWF:ready:descriptive-metadata',1,'accessionWF:ready:sdr-ingest-received',1,'accessionWF:waiting',1,'accessionWF:waiting:content-metadata',1,'accessionWF:waiting:descriptive-metadata',1,'accessionWF:waiting:end-accession',1,'accessionWF:waiting:provenance-metadata',1,'accessionWF:waiting:publish',1,'accessionWF:waiting:remediate-object',1,'accessionWF:waiting:reset-workspace',1,'accessionWF:waiting:rights-metadata',1,'accessionWF:waiting:sdr-ingest-received',1,'accessionWF:waiting:sdr-ingest-transfer',1,'accessionWF:waiting:shelve',1,'accessionWF:waiting:technical-metadata',1],
      'wf_swp_ssim' => ['completed',1,'completed:accessionWF',1,'completed:accessionWF:start-accession',1,'ready',1,'ready:accessionWF',1,'ready:accessionWF:descriptive-metadata',1,'ready:accessionWF:sdr-ingest-received',1,'waiting',1,'waiting:accessionWF',1,'waiting:accessionWF:content-metadata',1,'waiting:accessionWF:descriptive-metadata',1,'waiting:accessionWF:end-accession',1,'waiting:accessionWF:provenance-metadata',1,'waiting:accessionWF:publish',1,'waiting:accessionWF:remediate-object',1,'waiting:accessionWF:reset-workspace',1,'waiting:accessionWF:rights-metadata',1,'waiting:accessionWF:sdr-ingest-received',1,'waiting:accessionWF:sdr-ingest-transfer',1,'waiting:accessionWF:shelve',1,'waiting:accessionWF:technical-metadata',1],
      'has_model_ssim' => ['info:fedora/dor:googleScannedBookWF',23,'info:fedora/afmodel:Dor_Item',8,'info:fedora/dor:googleScannedBook',8,'info:fedora/afmodel:Dor_AdminPolicyObject',7,'info:fedora/afmodel:Dor_WorkflowObject',2,'info:fedora/afmodel:Dor_Collection',1],
      'metadata_source_ssi' => ['DOR',15,'Symphony',3],
      'registered_dttsim' => [],
      'accessioned_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'published_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'submitted_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'deposited_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'modified_latest_dttsi' => ['2016-02-18T23:15:12Z',2,'2016-02-18T23:15:22Z',2,'2016-02-18T23:15:01Z',1,'2016-02-18T23:15:05Z',1,'2016-02-18T23:15:10Z',1,'2016-02-18T23:15:15Z',1,'2016-02-18T23:15:16Z',1,'2016-02-18T23:15:17Z',1,'2016-02-18T23:15:18Z',1,'2016-02-18T23:15:19Z',1],
      'opened_latest_dttsi' => [],'embargo_release_dtsim' => ['2259-02-24T00:00:00Z',1],'sw_format_ssim' => ['Image',3,'Book',2],
      'sw_pub_date_facet_ssi' => ['1806',1,'1975',1,'2011',1,'2013',1],'topic_ssim' => ['Cephalopoda',1,'French spoliation claims',1,'agreement',1],
      'sw_subject_geographic_ssim' => ['Bermuda Islands',1,'North Pacific Ocean',1,'United States Pennsylvania Philadelphia',1],
      'sw_subject_temporal_ssim' => [],
      'sw_genre_ssim' => ['Bibliography',1,'Internal document',1],
      'sw_language_ssim' => ['English',3],
      'mods_typeOfResource_ssim' => ['still image',3,'text',3,'other',1],
      'exploded_tag_ssim' => ['Project',12,'Remediated By',10,'Registered By',8,'Project : DOR',6,'Remediated By : 3.25.3',6,'Project : Fuller Slides',3,'Registered By : renzo',3,'Remediated By : 4.6.6.2',3,'Process',2,'Process : Content Type',2,'Process : Content Type : Book (flipbook, ltr)',2,'Project : Argo Demo',2,'Registered By : llam813',2,'DPG',1,'DPG : Curator Request',1,'DPG : Curator Request : Kasianovitz',1,'DPG : State Banking Commission Annual Reports',1,'DPG : State Banking Commission Annual Reports : Virginia',1,'Process : Rights',1,'Process : Rights : world',1,'Project : State Banking Commission Annual Reports',1,'Registered By : labware',1,'Registered By : mbklein',1,'Registered By : sunetid',1,'Remediated By : 4.22.3',1],
      'objectType_ssim' => ['adminPolicy',7,'item',6,'agreement',2,'workflow',2,'collection',1],
      'content_type_ssim' => ['image',3,'file',2],
      'rights_primary_ssi' => ['dark',13,'stanford',3,'world',2],
      'use_license_machine_ssi' => [],
      'nonhydrus_collection_title_ssim' => ['Annual report of the State Corporation Commission showing the condition of the incorporated state banks and other institutions operating in Virginia at the close of business',1],
      'hydrus_collection_title_ssim' => [],
      'nonhydrus_apo_title_ssim' => ['Stanford University Libraries - Special Collections',4,'State Banking Commission Annual Reports',2,'[Internal System Objects]',12],
      'hydrus_apo_title_ssim' => [],'current_version_isi' => ['1',11,'2',4,'4',1],
      'processing_status_text_ssi' => ['Unknown Status',15,'In accessioning',1],
      'released_to_ssim' => [],
      'wf_wps_ssim' => ['accessionWF',1,'accessionWF:content-metadata',1,'accessionWF:content-metadata:waiting',1,'accessionWF:descriptive-metadata',1,'accessionWF:descriptive-metadata:ready',1,'accessionWF:descriptive-metadata:waiting',1,'accessionWF:end-accession',1,'accessionWF:end-accession:waiting',1,'accessionWF:provenance-metadata',1,'accessionWF:provenance-metadata:waiting',1,'accessionWF:publish',1,'accessionWF:publish:waiting',1,'accessionWF:remediate-object',1,'accessionWF:remediate-object:waiting',1,'accessionWF:reset-workspace',1,'accessionWF:reset-workspace:waiting',1,'accessionWF:rights-metadata',1,'accessionWF:rights-metadata:waiting',1,'accessionWF:sdr-ingest-received',1,'accessionWF:sdr-ingest-received:ready',1,'accessionWF:sdr-ingest-received:waiting',1,'accessionWF:sdr-ingest-transfer',1,'accessionWF:sdr-ingest-transfer:waiting',1,'accessionWF:shelve',1,'accessionWF:shelve:waiting',1,'accessionWF:start-accession',1,'accessionWF:start-accession:completed',1,'accessionWF:technical-metadata',1,'accessionWF:technical-metadata:waiting',1],
      'wf_wsp_ssim' => ['accessionWF',1,'accessionWF:completed',1,'accessionWF:completed:start-accession',1,'accessionWF:ready:descriptive-metadata',1,'accessionWF:ready:sdr-ingest-received',1,'accessionWF:waiting',1,'accessionWF:waiting:content-metadata',1,'accessionWF:waiting:descriptive-metadata',1,'accessionWF:waiting:end-accession',1,'accessionWF:waiting:provenance-metadata',1,'accessionWF:waiting:publish',1,'accessionWF:waiting:remediate-object',1,'accessionWF:waiting:reset-workspace',1,'accessionWF:waiting:rights-metadata',1,'accessionWF:waiting:sdr-ingest-received',1,'accessionWF:waiting:sdr-ingest-transfer',1,'accessionWF:waiting:shelve',1,'accessionWF:waiting:technical-metadata',1],
      'wf_swp_ssim' => ['completed',1,'completed:accessionWF',1,'completed:accessionWF:start-accession',1,'ready',1,'ready:accessionWF',1,'ready:accessionWF:descriptive-metadata',1,'ready:accessionWF:sdr-ingest-received',1,'waiting',1,'waiting:accessionWF',1,'waiting:accessionWF:content-metadata',1,'waiting:accessionWF:descriptive-metadata',1,'waiting:accessionWF:end-accession',1,'waiting:accessionWF:provenance-metadata',1,'waiting:accessionWF:publish',1,'waiting:accessionWF:remediate-object',1,'waiting:accessionWF:reset-workspace',1,'waiting:accessionWF:rights-metadata',1,'waiting:accessionWF:sdr-ingest-received',1,'waiting:accessionWF:sdr-ingest-transfer',1,'waiting:accessionWF:shelve',1,'waiting:accessionWF:technical-metadata',1],
      'has_model_ssim' => ['info:fedora/dor:googleScannedBookWF',23,'info:fedora/afmodel:Dor_Item',8,'info:fedora/dor:googleScannedBook',8,'info:fedora/afmodel:Dor_AdminPolicyObject',7,'info:fedora/afmodel:Dor_WorkflowObject',2,'info:fedora/afmodel:Dor_Collection',1],
      'metadata_source_ssi' => ['DOR',15,'Symphony',3],
      'registered_dttsim' => [],
      'accessioned_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'published_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'submitted_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'deposited_latest_dttsi' => ['2016-02-22T22:35:41Z',1],
      'modified_latest_dttsi' => ['2016-02-18T23:15:12Z',2,'2016-02-18T23:15:22Z',2,'2016-02-18T23:15:01Z',1,'2016-02-18T23:15:05Z',1,'2016-02-18T23:15:10Z',1,'2016-02-18T23:15:15Z',1,'2016-02-18T23:15:16Z',1,'2016-02-18T23:15:17Z',1,'2016-02-18T23:15:18Z',1,'2016-02-18T23:15:19Z',1],
      'opened_latest_dttsi' => [],
      'embargo_release_dtsim' => ['2259-02-24T00:00:00Z',1],
      'sw_format_ssim' => ['Image',3,'Book',2],
      'sw_pub_date_facet_ssi' => ['1806',1,'1975',1,'2011',1,'2013',1],
      'topic_ssim' => ['Cephalopoda',1,'French spoliation claims',1,'agreement',1],
      'sw_subject_geographic_ssim' => ['Bermuda Islands',1,'North Pacific Ocean',1,'United States Pennsylvania Philadelphia',1],
      'sw_subject_temporal_ssim' => [],
      'sw_genre_ssim' => ['Bibliography',1,'Internal document',1],
      'sw_language_ssim' => ['English',3],
      'mods_typeOfResource_ssim' => ['still image',3,'text',3,'other',1]
    },
```

**Comment by atz** [comment]
Note the values appear to be the same... but this should be avoided, especially if the facets are being computed twice!

**Comment by cbeer** [comment]
I vaguely remember this is a problem with some versions of solr, maybe? Can you give https://github.com/projecthydra/hydra-jetty/tree/7.x-stable a try?

**Comment by atz** [comment]
It's already on `7.2.0`

**Comment by cbeer** [comment]
`7.x-stable` has Solr 4.10.4, which may make a difference.

**Comment by atz** [comment]
Runs clean through rspec.  Still have to investigate whether it solves this problem.  Should we make a new `7.3.0` release of hydra-jetty or don't bother?

**Comment by cbeer** [comment]
If you could confirm whether it fixes all those deprecation warnings, yes, let's cut a release.

**Comment by mejackreed** [comment]
Closed by #607
